File: DotNet/InboundAdapter.cs
using System;

namespace InterSystems.Demo.PEX
{
    class InboundAdapter : InterSystems.EnsLib.PEX.InboundAdapter
    {
        public int min = 0;
        public int max = 0;
        int runningCount = 0;

        public override void OnInit()
        {
            Console.Write("\r\n[.NET]...Demo.PEX.InboundAdapter:OnInit() is called, " + " min=" + min + ", max=" + max);
            return;
        }

        public override void OnTearDown()
        {
            Console.Write("\r\n[.NET]...Demo.PEX.InboundAdapter:OnTearDown() is called");
            return;
        }

        public override void OnTask()
        {
            Console.Write("\r\n[.NET]...Demo.PEX.InboundAdapter:OnTask() is called");
            // 50% chance of calling ProcessInput()
            if ((new Random()).NextDouble() < 0.5)
            {
                SimpleObject request = new SimpleObject("message #" + (++runningCount));
                Console.Write("\r\n[.NET]...Demo.PEX.InboundAdapter:OnTask() calls ProcessInput() with " + request.value);
                string response = (string)BusinessHost.ProcessInput(request);
                Console.Write("\r\n[.NET]...Demo.PEX.InboundAdapter:OnTask() receives \"" + response + "\"");
            }
            return;
        }

    }
}

File: Java/src/main/java/com/intersystems/demo/pex/InboundAdapter.java
package com.intersystems.demo.pex;

public class InboundAdapter extends com.intersystems.enslib.pex.InboundAdapter {
	
	public int min = 0;
	public int max = 0;
	int runningCount = 0;
	
	public void OnInit() throws Exception {
		System.out.print("\r\n[Java] ...demo.pex.InboundAdapter:OnInit() is called, "+" min="+min+", max="+max);
		return;
	}

	public void OnTearDown() throws Exception {
		return;
	}

	public void OnTask() throws Exception {
		// 50% chance of calling ProcessInput()
		if ( Math.random()<0.5 ) {
			SimpleObject request = new SimpleObject("message #"+(++runningCount));
			System.out.print("\r\n[Java] ...demo.pex.InboundAdapter:OnTask() calls ProcessInput() with " + request.value);
			String response = (String) BusinessHost.ProcessInput(request);
			System.out.print("\r\n[Java] ...demo.pex.InboundAdapter:OnTask() receives \"" + response + "\"");
		}
		return;
	}

}
